update/files/multiFileReader.go

```diff
@@ -0,0 +1,124 @@
+package files
+
+import (
+	"bufio"
+	"io"
+	"io/ioutil"
+	"os"
+	"sort"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/update"
+)
+
+type multiFileReader struct {
+	importFolder  string
+	files         map[string]*os.File
+	dataReader    map[string]*bufio.Scanner
+	importStorage storage.Storer
+}
+
+type ArgsNewMultiFileReader struct {
+	ImportFolder  string
+	ImportStorage storage.Storer
+}
+
+func NewMultiFileReader(args ArgsNewMultiFileReader) (*multiFileReader, error) {
+	if len(args.ImportFolder) < 2 {
+		return nil, update.ErrInvalidFolderName
+	}
+	if check.IfNil(args.ImportStorage) {
+		return nil, update.ErrNilStorage
+	}
+
+	m := &multiFileReader{
+		importFolder:  args.ImportFolder,
+		files:         make(map[string]*os.File),
+		dataReader:    make(map[string]*bufio.Scanner),
+		importStorage: args.ImportStorage,
+	}
+
+	err := m.readAllFiles()
+	if err != nil {
+		return nil, err
+	}
+
+	return m, nil
+}
+
+func (m *multiFileReader) readAllFiles() error {
```

Comment:
wondering how we should test this

Reply:
sometimes it is important to not mock everything. Creating folder and files, reading from the one created.

Reply:
Agree, that will be tested in integration tests. Unit tests should be able to test with mocks and stubs and reach 100% coverage (ideally).